docs/architecture/adr/adr-006-in-cluster-db.md

```diff
@@ -0,0 +1,115 @@
+# ADR 006: In-cluster Database
+
+## Context
+
+Microsoft is sunsetting Azure Database for MariaDB summer 2025.
+They recommend moving to Azure Database for MySql instead.
+
+The project has experienced project wide down time due to database crashes on
+multiple occasions:
+
+- 17.01.2025: Database went down due to a restart initiated by the Azure Product
+Group team taking unsanctioned action
+- 15.01.2025: Database went down due to a restart initiated by the Azure Product
+Group team taking unsanctioned action
+- 9.01.2025: Database went down as a result of to many login requests
+- 28.11.2024: Database went down due to too many active connection attempts
+- 7.11.2024: Database went down due to too many connections
+- Summer 2024: Database was restarted by Azure for unscheduled and unnotified
+maintenance
+
+Microsoft's database server logs are prohibitively expensive.
+Microsoft misconfigured the database provision-time.
+The current Microsoft support tier is useless and useful support is
+prohibitively expensive.
+Microsoft support has made unsanctioned changes on the databse, such as
+maintenance work during business hours and restarts in attempts to fix support
+tickets.
+It is not possible to access the database server.
+
+Installing a MariaDB Sql operator in the cluster provides the following
+benefits:
+
+- access to the database server logs
+- access to the server itself
+- a guarantee that no one but the platform team can take action on the server
+- ability to right-size the database
+- Running clusters of databases
+- Database recovery
+- The ability to start recovering from a crash immediately
+
+### Pros & Cons
+
+#### Azure Managed DB
+
+##### Cons
+
+- Experienced to be poorly managed
+- Server logs is disproportionately priced = too expensive for this project
+- We don't have access to the server
+- The wait time for support can be long and incorrect.
+- Server might be (was in our case) misconfigured
+- Will require a migration to a potentially as bad service
+
+##### Pros
+
+- One-click setup
+- Minimal configuration needed from us
+- Azure support
+- Azure can be blamed for downtime, they are directly or indirectly responsible
+for.
+- MySql, which is not set to be sunset, might have a noticably better service
+as that is their chosen flavor to keep offering.
```

Comment:
I think all this can be reduced to something like "Continual use of managed database product".

Reply:
all of the bullet points?